xds/src/main/java/io/grpc/xds/XdsComms.java

```diff
@@ -239,7 +238,8 @@ public void run() {
                   localityStore.updateDropPercentage(dropOverloads);
 
                   List<LocalityLbEndpoints> localities = clusterLoadAssignment.getEndpointsList();
-                  Map<XdsLocality, LocalityInfo> localityEndpointsMapping = new LinkedHashMap<>();
+                  ImmutableMap.Builder<XdsLocality, LocalityInfo> localityEndpointsMappingBuilder =
```

Comment:
Why do this inverse change?

Reply:
You are getting rid of `ImmutableMap.Builder` pattern. But this change changes from `Collections.unmodifiableMap(...)` to using `ImmutableMap.Builder`.

Reply:
Did I? I don't see it was ever `ImmutableMap.Builder` before. https://github.com/grpc/grpc-java/commit/213b91b16583b0a982c8c32efbbb3f2a2406356c#diff-9e293ee2079eae78a4ffd73aeb825704R221